### PR TITLE
Beta version

### PR DESCRIPTION
**Pull request by ujja**
Integration test is not done, botmock have to implement txt export, and had to change typescript schema first

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (szabi84) [inline tsconfig.json:1]: I would put these compiler options as well:
`   "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitAny": true, `

File: src/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+export interface BotiumHeader {
```
Comment (szabi84) [inline src/types/index.ts:1]: This interface is not used anywhere

File: src/BotiumBotmockExporter.ts [hunk 3]
```diff
@@ -0,0 +1,443 @@
+import "dotenv/config";
+import {BaseExporter, Resources, FileWriter, Kind, ProjectReference} from "@botmock/export";
```
Comment (szabi84) [inline src/BotiumBotmockExporter.ts]: With the options that I mentioned at tsconfing.json you get in this file some unused import and variable.

File: src/BotiumBotmockExporter.ts [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+    #intentToUtteranceRef = (intent: string): string => {
+        return `UTT_${intent.split(' ').join('_').toUpperCase()}`
+    }
+
+    /**
+     * Replaces botmock variables in text with botium variables
+     * @param text Text in block
+     */
+    #replaceVariableCharacterInTextDeprecated = (text: string): string => {
+        const alphanumericRegexp = new RegExp(/\%([a-zA-Z0-9_]+)\%/g);
```
Comment (szabi84) [inline src/BotiumBotmockExporter.ts:82]: The \ is not needed before %:
```suggestion
        const alphanumericRegexp = new RegExp(/%([a-zA-Z0-9_]+)%/g);
```

File: src/BotiumBotmockExporter.ts [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+        for (const match of matches as string[]) {
+            const variableNameInsideMatch = match.substr(1, match.length - 2);
+            replacedText = replacedText.replace(match, `$${variableNameInsideMatch}`);
+        }
+        return replacedText;
+    };
+
+    /**
+     * Replaces botmock variables in text with botium variables
+     * @param text Text in block
```
Comment (szabi84) [inline src/BotiumBotmockExporter.ts:97]: The variables param is missing form JsDoc.

File: src/BotiumBotmockExporter.ts [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+            const convo: Botium.Convo = {
+                header: {
+                    name: convoGeneratedName,
+                    description: convoGeneratedDescription
+                },
+                conversation: botiumConvoSteps
+            }
+
+            const data = compiler.Decompile([convo], SCRIPTING_FORMAT_TXT, SCRIPTING_TYPE_CONVO)
+            var fs = require('fs');
```
Comment (szabi84) [inline src/BotiumBotmockExporter.ts]: I think fs could be imported globally in the file's header:
`import { writeFileSync } from 'fs'`
You need to install @types/node first: npm install @types/node --save-dev

File: src/BotiumBotmockExporter.ts [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+import {BotDriver} from "botium-core"
+import {sanitize} from "sanitize-filename-ts";
+
+const SCRIPTING_FORMAT_TXT = 'SCRIPTING_FORMAT_TXT'
+const SCRIPTING_TYPE_CONVO = 'SCRIPTING_TYPE_CONVO'
+const SCRIPTING_TYPE_UTTERANCES = 'SCRIPTING_TYPE_UTTERANCES'
+
+import {Botium} from "./types";
+
+interface Log {
```
Comment (szabi84) [inline src/BotiumBotmockExporter.ts:12]: In my opinion these interfaces could be moved into 'types/index.ts' as well.

Comment (ujja) [reply src/BotiumBotmockExporter.ts:12]: Those are internal interfaces. types/index.ts are external. Is this not correct to separate those this way?

Comment (szabi84) [reply src/BotiumBotmockExporter.ts:12]: I saw this kind of code organization in typescript, that the types were defined in a separate file. But it's absolutely not critical.

**Pull request merged**